### ISSUE TITLE
Tile selection weirdness

### ISSUE THREAD

**Issue by Enrico-Swagolo** [LoneGazebo/Community-Patch-DLL#2876]
_1. Mod version (i.e Date - 4/23):_
Current

_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description:_ 
Basically, tile selection AI sometimes ignores second ring luxuries. I didn't see it ignore bonus resources or strategics as frequently (if at all), but an olive on 2nd row was ignored in favour of plain, grassland and forest grassland. It doesn't always happen though.

Also AI really likes tiles with bonus resources on them, even when they're on a desert. Still, I can get that - stone on a desert turns into a 5 prod tile pretty soon which is not too shabby by itself. It'd be better if the tile was ignored in favour of a grassland bonus resource though, and it wasn't.

Overall, besides those the tile selection is more reliable than previously so it's a good job. It's just that sometimes, for some reason, it decides to be weird.

**Comment by LoneGazebo** [comment]
@ilteroi I told you not to mess with it - now you're the problem! :)

**Comment by ricanuck** [comment]
Are we sure we want 100% predictability here?

**Comment by LoneGazebo** [comment]
@ricanuck no, but the selector is making actively aberrant choices in some situations. It should be rational (i.e. not make players think it is doing the wrong thing).

**Comment by Radost** [comment]
Doesn't look very rational to me:
![8930_20161207214138_1](https://cloud.githubusercontent.com/assets/1131228/20984419/ef4a7d76-bce1-11e6-88e4-070c5fb59043.png)
Note the two wheat tiles, a horses tile and a forest tile.

**Comment by ilteroi** [comment]
@Radost : are you using CP only or the full VP?

**Comment by Enrico-Swagolo** [comment]
I see similar stuff as Radosct and I use full VP. No changes besides insta resource reveal which doesn't affect it.

I don't even get why it sometimes ignores resources but sometimes it's absolutely obsessed about them, primarily if they're on the desert (usually goes to them first)

**Comment by Radost** [comment]
@ilteroi 
Full VP

**Comment by ilteroi** [comment]
hmm. i did find a bug, don't know if it can explain the behavior with the "holes" though.

care to test with this dll? [CvGameCore_Expansion2.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/640604/CvGameCore_Expansion2.zip)

**Comment by Enrico-Swagolo** [comment]
Another weird situation - (prev dll) - the nwonders aren't even valued properly. The tile selection thingy thought mines of salomon are equal to a forested tundra. It still gave me the mines, but the pink selector was also on tundra forests.


Will try the new one, is it compatible with old games? This Rome game seems interesting.

**Comment by ilteroi** [comment]
well, both being highlighted doesn't mean they have equal value, only that both are among the top 3.

is savegame compatible.

**Comment by Enrico-Swagolo** [comment]
Hm,  with the new DLL I still see NW and 2 other regular tiles being highlighted, but every time this has happened, the city decided to get the NW so I guess while it shows them as equally valued by the city, they aren't. Or RNG, it's possible, my sample size is like 3 NWs. 

I also saw multiple CSes not expanding to  their adjacent luxuries, instead opting to take plains/grass/tundra usually with forests despite both being in the 2nd row. It seems more reliable though than pre-DLL, but I've only played 2 games (ended both at classical). It seems to be valuing strategics and luxuries perfectly sometimes (my city up north properly decided that 2 iron tiles and salt near it were worthwhile and marked them), but sometimes it doesn't.
